integrations/huggingface-transformers/tutorials/sparse-transfer-learning-bert.md

```diff
@@ -0,0 +1,343 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Sparse Transfer Learning
+
+This page explains how to fine-tune a pre-sparsified BERT model onto a downstream dataset with SparseML.
+
+## Overview
+
+Sparse Transfer is quite similiar to the typical NLP transfer learning, where we fine-tune a checkpoint pretrained on a large dataset like Wikipedia BookCorpus onto a smaller downstream dataset and task. However, with Sparse Transfer Learning, we simply start the fine-tuning process from a pre-sparsified model and maintain sparsity while the training process occurs.
+
+[SparseZoo](https://sparsezoo.neuralmagic.com/?domain=nlp&sub_domain=masked_language_modeling&page=1), Neural Magic's open-source Model Zoo, contains pre-sparsified checkpoints of common NLP models like BERT-base, BERT-large, and RoBERTa. These models can be used as the starting checkpoint for the sparse transfer learning workflow.
+
+## Installation
+
+Install via `pip`:
+
+```
+pip install sparseml[torch]
+```
+
+## Sparse Transfer Learning onto SST2
+
+Let's try a simple example of fine-tuning a pre-sparsified model onto the SST dataset. SST2 is a sentiment analysis
+dataset, with pairs of sentences and binary labels representing positive or negative sentiment.
+
+[SST2 Dataset Card](https://huggingface.co/datasets/glue/viewer/sst2/train)
+
+### Selecting a Pre-sparsified Upstream Model
+
+We will fine-tune a [90% sparsified BERT-base](https://arxiv.org/abs/2203.07259). This model is hosted in [SparseZoo](https://sparsezoo.neuralmagic.com/models/nlp%2Fmasked_language_modeling%2Fobert-base%2Fpytorch%2Fhuggingface%2Fwikipedia_bookcorpus%2Fpruned90-none) and is identified by the following stub:
+```bash
+zoo:nlp/masked_language_modeling/obert-base/pytorch/huggingface/wikipedia_bookcorpus/pruned90-none
+```
+
+SparseML uses the stub to download the model before starting training.
+
+### Kick off Training
+
+Run the following to fine-tune the pre-sparsified BERT onto SST2:
+
+```bash
+sparseml.transformers.text_classification \
+  --output_dir pruned_quantized_obert-text_classification_sst2 \
+  --model_name_or_path zoo:nlp/masked_language_modeling/obert-base/pytorch/huggingface/wikipedia_bookcorpus/pruned90-none \
+  --recipe zoo:nlp/sentiment_analysis/obert-base/pytorch/huggingface/sst2/pruned90_quant-none \
+  --distill_teacher zoo:nlp/sentiment_analysis/obert-base/pytorch/huggingface/sst2/base-none \
+  --task_name sst2 --max_seq_length 128 --per_device_train_batch_size 32 --per_device_eval_batch_size 32 --preprocessing_num_workers 6 \
+  --do_train --do_eval --evaluation_strategy epoch --fp16  \
+  --save_strategy epoch --save_total_limit 1
+```
+
+Lets discuss the key arguments:
+
+- `--task` specifies the dataset to train on (in this case SST2). You can pass any GLUE task to the `--task` command. Check out the use case pages for details on passing a custom dataset.
```

Comment:
In the code above we have `task_name`. Should they be the same?

Reply:
`Check out the use case pages for details on passing a custom dataset`. Where are these pages?

Reply:
in this case, SST2